libcontainer/userns/userns_maps_linux.go

```diff
@@ -169,3 +169,18 @@ func GetUserNamespaceMappings(nsPath string) (uidMap, gidMap []configs.IDMap, er
 
 	return uidMap, gidMap, nil
 }
+
+// IsSameMapping returns whether or not the two id mappings are the same. Note
+// that if the order of the mappings is different, or a mapping has been split,
+// the mappings will be considered different.
+func IsSameMapping(a, b []configs.IDMap) bool {
```

Comment:
Can we just use `reflect.DeepEqual` ?

Reply:
Reflection is slower and importing `reflect` IIRC means the compiler can't do certain optimisations. We do use `reflect` elsewhere in runc, but it seems best to not add more uses that aren't actually necessary.

Once we switch to Go 1.21, we can use `slices.Equal`. I would prefer that over either option.